module/VuFind/src/VuFind/Db/Service/AuthHashService.php

```diff
@@ -97,7 +117,21 @@ public function getByHashAndType(string $hash, string $type, bool $create = true
      */
     public function getLatestBySessionId(string $sessionId): ?AuthHashEntityInterface
     {
-        return $this->getDbTable('AuthHash')->getLatestBySessionId($sessionId);
+        $dql = 'SELECT ah '
+            . 'FROM ' . $this->getEntityClass(AuthHash::class) . ' ah '
+            . 'WHERE ah.sessionId = :sessionId '
+            . 'ORDER BY ah.created DESC';
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameter('sessionId', $sessionId);
+        $result = $query->getOneOrNullResult();
+        if ($result === null) {
```

Comment:
The original Laminas version did not create a row if none was found; it just returned null.